src/formatDate.js

```diff
@@ -7,8 +7,50 @@
  *
  * @returns {string}
  */
-function formatDate(date, fromFormat, toFormat) {
-  // write code here
+function formatDate(date, sourceFormat, targetFormat) {
+  // Делаем деструктуризацию кода
+  const [, , , sourceSeparator] = sourceFormat;
+  const [, , , targetSeparator] = targetFormat;
+
+  // Разделение даты по сепаратору
+  const dateParts = date.split(sourceSeparator);
+  const [sourceYear, sourceMonth, sourceDay] = dateParts;
+
+  const formattedDateParts = [];
+
+  // Переформатирование даты в зависимости от исходного формата
+  if (
+    sourceFormat[0] === 'YYYY' &&
+    targetFormat[0] === 'DD' &&
+    targetFormat[1] === 'MM' &&
+    targetFormat[2] === 'YYYY'
```

Comment:
The condition checks for a specific source and target format, but it does not handle other possible combinations. You should generalize the condition to work with any given format, not just the hardcoded 'YYYY', 'DD', 'MM' sequence.